nixops/backends/__init__.py

```diff
@@ -50,14 +51,20 @@ def __init__(self, name: str, config: nixops.resources.ResourceEval):
 class MachineState(nixops.resources.ResourceState):
     """Base class for NixOps machine state objects."""
 
+    defn: Optional[MachineDefinition]
```

Comment:
Should `defn` and its setting, be part of `ResourceState`'s default implementation?

Reply:
Yes, I think so.